source/simulator/assembly.cc

```diff
@@ -1701,6 +1807,839 @@ namespace aspect
+                scratch.face_current_velocity_values);
+
+            //get the mesh velocity, as we need to subtract it off of the advection systems
+            if (parameters.free_surface_enabled)
+              scratch.face_finite_element_values[introspection.extractors.velocities].get_function_values(free_surface->mesh_velocity,
+                  scratch.face_mesh_velocity_values);
+
+            compute_material_model_input_values (current_linearization_point,
+                                                 scratch.face_finite_element_values,
+                                                 cell,
+                                                 true,
+                                                 scratch.face_material_model_inputs);
+            material_model->evaluate(scratch.face_material_model_inputs,
+                                     scratch.face_material_model_outputs);
+
+            HeatingModel::HeatingModelOutputs face_heating_model_outputs(n_q_points, parameters.n_compositional_fields);
+            heating_model_manager.evaluate(scratch.face_material_model_inputs,
+                                           scratch.face_material_model_outputs,
+                                           face_heating_model_outputs);
+
+            if (((parameters.fixed_temperature_boundary_indicators.find(cell->face(f)->boundary_id()) != parameters.fixed_temperature_boundary_indicators.end()) && (advection_field.is_temperature()))
+                ||
+                (!advection_field.is_temperature()))
+              {
+                //We are either in the case of a Dirichlet temperature boundary, or we are assembling for a composition variable. In the former, impose the Dirichlet value weakly using a matrix term and RHS term. In the latter, if this is an inflow boundary then impose a zero Dirichlet value by assembling the corresponding matrix term but no RHS term
+
+                for (unsigned int q=0; q<n_q_points; ++q)
+                  {
+                    // precompute the values of shape functions and their gradients.
+                    // We only need to look up values of shape functions if they
+                    // belong to 'our' component. They are zero otherwise anyway.
+                    // Note that we later only look at the values that we do set here.
+                    for (unsigned int k=0; k<advection_dofs_per_cell; ++k)
+                      {
+                        scratch.face_grad_phi_field[k]      = scratch.face_finite_element_values[solution_field].gradient (scratch.face_finite_element_values.get_fe().component_to_system_index(solution_component, k), q);
+                        scratch.face_phi_field[k]      = scratch.face_finite_element_values[solution_field].value (scratch.face_finite_element_values.get_fe().component_to_system_index(solution_component, k), q);
```

Comment:
Please align the lines or just separate everything with one space on either side of the equal sign.